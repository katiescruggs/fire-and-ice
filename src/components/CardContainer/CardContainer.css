.card-container {
  display: flex;
  flex-wrap: wrap;
  margin: 0 auto;
  width: 1032px;
}

@media screen and (max-width: 1180px) {
  .card-container {
    width: 688px;
  }
}

@media screen and (max-width: 880px) {
  .card-container {
    width: 344px;
  }
}