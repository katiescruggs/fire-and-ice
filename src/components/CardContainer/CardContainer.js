import React from 'react';
import Card from '../Card/Card';
import { connect } from 'react-redux';
import './CardContainer.css';

const CardContainer = ({ houses }) => {
  const cards = houses.map(house => <Card house={house} />);

  return (
    <div className="card-container">
      {cards}
    </div>
  );
}

const mapStateToProps = ({ houses }) => ({
  houses
});

export default connect(mapStateToProps, null)(CardContainer);