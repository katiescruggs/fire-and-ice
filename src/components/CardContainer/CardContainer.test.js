import React from 'react';
import { shallow } from 'enzyme';
import { CardContainer, mapStateToProps } from './CardContainer';

describe('CardContainer Container', () => {
  describe('CardContainer', () => {
    let cardContainer;
    const mockHouses = ['array', 'of', 'houses'];

    beforeEach(() => {
      cardContainer = shallow(<CardContainer houses={mockHouses} />);
    });

    it('should exist', () => {
      expect(cardContainer).toBeDefined();
    });

    it('should match snapshot', () => {
      expect(cardContainer).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('connects houses to props', () => {
      const mockStore = {
        houses: ['array', 'of', 'houses']
      };

      const result = mapStateToProps(mockStore);

      expect(result.houses).toEqual(mockStore.houses);
    });
  });
});