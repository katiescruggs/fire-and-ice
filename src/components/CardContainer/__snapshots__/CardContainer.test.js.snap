// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer Container CardContainer should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    houses={
        Array [
            "array",
            "of",
            "houses",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          house="array"
/>,
        <Card
          house="of"
/>,
        <Card
          house="houses"
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "card-0",
        "nodeType": "function",
        "props": Object {
          "house": "array",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "card-1",
        "nodeType": "function",
        "props": Object {
          "house": "of",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "card-2",
        "nodeType": "function",
        "props": Object {
          "house": "houses",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            house="array"
/>,
          <Card
            house="of"
/>,
          <Card
            house="houses"
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "card-0",
          "nodeType": "function",
          "props": Object {
            "house": "array",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "card-1",
          "nodeType": "function",
          "props": Object {
            "house": "of",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "card-2",
          "nodeType": "function",
          "props": Object {
            "house": "houses",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
